contrib/nix/util/readDirBySuffix.nix

```diff
@@ -0,0 +1,17 @@
+let
+  inherit (builtins) attrNames listToAttrs readDir;
+  inherit (import ./nixpkgs.nix) lib;
```

Comment:
General question: in most of the nix code I've interacted with previously, it doesn't seem to take this approach of using `import`, but usually expects everything to be passed in as arguments instead (possibly supplying defaults). However, that does clutter up call sites, particularly for this kind of utility code. I'm not sure what the real "nix way" is here - is the way you've done it here a pattern you've observed?